### DISCUSSION TITLE
GNSS

### DISCUSSION THREAD

**Discussion by sygandan** [hatlabs/discussions#52] [Q&A]
Hello

Got stuck installing MAX-M8Q and have tried everything I found on internet.
The hat is stacked on top on CM4-WS485-WSCAN-SHRPi. GPIO-pin cutted as instructed.
Ran automated installation script without any problems and have tried with different combination enabling/disabling the other hats.
The GNSS is seen as device fe201000.serial ttyAMA0 with alias ttyOP_gnss, connection over NMEA 0183/SignalK at 9600 bauds (even tried 4800 and 38400). Everything seems to be ok and SK network in CPN is enabled (localhost, port 3000). LEDs on HAT shows power on, PPS and Txd flashing approximatley every second. Yet no signal in CPN.
Tried another USB GPS puck over serial data connection with no problems.

This begins to be rather frustrating and any help on how to proceed would be very appreciated!

**Comment by sygandan** [comment]
Using cat/dev/ttyAMA0 get: "State: ON, V_sup: 967, V_in: 378, I_in: 119, temp: 39088, i2c: 255, RTC: 1, EXT: 0, PWR: 0"

I suppose thats the status from SH-RPi HAT, and not the GNSS HAT so obviously there is something I have missed.

**Reply by mairas** [reply]
The string you quoted is indeed SH-RPi console output. I suppose you have the jumper caps on the header?

The instructions are indeed a bit misleading. The GNSS serial port on a standard OpenPlotter setup is /dev/ttyS0. Try that; it is working for me.

**Reply by sygandan** [reply]
Yes, the two jumpers are on "B" pins. In the end I messed things up and reinstalled everything and suddenly ttyS0 went up by itself and GNSS works, but also need to have ttyAMA0 (fe201000.serial) connected.
! I think maybe I did something wrong with the GNSS HAT software installation the first time.

I didnt succeed flashing the SH-RPi new firmware by the way and dont dare enable BT again as it still says UART0 interface in GPIO 14 and 15 will be disabled. I suppose that means GNSS will disappear again? 
Got filenotfound errors in last step (megaavr).

**Comment by sygandan** [comment]
Funny but when I execute a check system it says /dev/ttyS0 has  no alias but as it never appeared as a device, just under connections, it isnt possible to give it an alias. Cant see it is needed neither?
